clippy_lints/src/methods/rewind_instead_of_seek_to_start.rs

```diff
@@ -0,0 +1,34 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::match_qpath;
+use rustc_ast::ast::{LitIntType, LitKind};
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::Span;
+
+use super::REWIND_INSTEAD_OF_SEEK_TO_START;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, arg: &'tcx Expr<'_>, name_span: Span) {
+    if_chain! {
+        if let ExprKind::Call(func, args1) = arg.kind;
+        if let ExprKind::Path(ref qpath1) = func.kind;
+        if match_qpath(qpath1, &["SeekFrom", "Start"]);
+        if args1.len() == 1;
+        if let ExprKind::Lit(ref lit) = args1[0].kind;
+        if let LitKind::Int(0, LitIntType::Unsuffixed) = lit.node;
```

Comment:
Can the check that receiver type implements `std::io::Seek` trait be added? The method below will be helpful:
https://github.com/rust-lang/rust-clippy/blob/502e87c3790059e878868125985726c065b72f85/clippy_utils/src/ty.rs#L150

Reply:
I implemented this suggestion by using `get_trait_def_id(cx, &paths::STD_IO_SEEK)`.
A different approach would have been to define a new symbol(`IoSeek`) similar to [https://github.com/rust-lang/rust/blob/master/compiler/rustc_span/src/symbol.rs#L216](url) and just use `get_diagnostic_item(sym::IoSeek)`, but that implied a change in the `rust` repo also, right?

Reply:
> but that would have implied a change in the rust repo also, right?

yes